algorithms/ar-knn/c/knn.c

```diff
@@ -0,0 +1,122 @@
+#include<stdio.h>
+#include<inttypes.h>
+#include<math.h>
+#include<stdlib.h>
+typedef struct{
+    float x;
+    float y;
+    int_fast8_t class_or_value;
+}Point;
+static float *distance = NULL;
+static unsigned k;
+static int_fast8_t *class_count = NULL;
+// utility function to calculate distance between two points using Euclidean Distance Method
+float get_euclidean_distance(const Point p, const Point q){
+    return sqrtf((p.x - q.x) * (p.x - q.x) + (p.y-q.y) * (p.y-q.y));
+}
+// update class or point values in training set
+// can be used for both regressor and classifier
+void assign_classes(Point training_set[], const int_fast8_t classes[], const size_t size){
+    for(int i = 0; i < (int)size; ++i){
+        training_set[i].class_or_value = classes[i];
+    }   
+}
+// calculate distance between each given point and test point
+void update_distance(const Point training_set[], const size_t size,const Point q){
+    distance = (float *)malloc(size * sizeof(float));
+    for(int i = 0; i < (int)size; ++i){
+        distance[i] = get_euclidean_distance(training_set[i], q); 
+    }
+}
+// utility function for swapping elements in collection
+void swap(void *x, void *y){
+    void *temp = x;
+    x = y;
+    y = temp;
```

Comment:
Codacy found an issue: [Assignment of function parameter has no effect outside the function. Did you forget dereferencing it?](https://app.codacy.com/gh/FOSS-UCSC/FOSSALGO/pullRequest?prid=6388877)